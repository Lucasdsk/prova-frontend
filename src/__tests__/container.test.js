import React from 'react'
import Container from './../components/container'

describe('<Container />', () => {
  
  it("Render Container", () => {
    const ContainerComponent = shallow(<Container />)
    expect(ContainerComponent).toMatchSnapshot()
  })

  it("Should render a children when it's passed", () => {
    const ContainerComponent = shallow(<Container>
      <p>Children</p>
    </Container>)
    expect(ContainerComponent.find('p').length).toBe(1)
  })

  describe("Test classNames according prop passed", () => {
    it("Shouldn't have class 'container--primary' when 'primary' prop is 'false", () => {
      const ContainerComponent = shallow(<Container primary={false} />)
      expect(ContainerComponent.hasClass('container--primary')).toBe(false)
    })

    it("Should have class 'container--fluid' when have 'fluid' prop", () => {
      const ContainerComponent = shallow(<Container fluid />)
      expect(ContainerComponent.hasClass('container--fluid')).toBe(true)
    })

    it("Should have class 'container--no-borders' when have 'noBorders' prop", () => {
      const ContainerComponent = shallow(<Container noBorders />)
      expect(ContainerComponent.hasClass('container--no-borders')).toBe(true)
    })

    it("Should have class 'container--no-border-top' when have 'noBorderTop' prop", () => {
      const ContainerComponent = shallow(<Container noBorderTop />)
      expect(ContainerComponent.hasClass('container--no-border-top')).toBe(true)
    })

    it("Should have class 'container--no-border-right' when have 'noBorderRight' prop", () => {
      const ContainerComponent = shallow(<Container noBorderRight />)
      expect(ContainerComponent.hasClass('container--no-border-right')).toBe(true)
    })

    it("Should have class 'container--no-border-bottom' when have 'noBorderBottom' prop", () => {
      const ContainerComponent = shallow(<Container noBorderBottom />)
      expect(ContainerComponent.hasClass('container--no-border-bottom')).toBe(true)
    })

    it("Should have class 'container--no-border-left' when have 'noBorderLeft' prop", () => {
      const ContainerComponent = shallow(<Container noBorderLeft />)
      expect(ContainerComponent.hasClass('container--no-border-left')).toBe(true)
    })

  })
})