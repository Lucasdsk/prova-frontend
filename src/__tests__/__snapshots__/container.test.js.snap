// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Container /> Render Container 1`] = `
<div
  className="container container--primary"
/>
`;
